pkgs/tools/misc/hdf4/default.nix

```diff
@@ -31,11 +31,12 @@ stdenv.mkDerivation rec {
     "-DHDF4_BUILD_WITH_INSTALL_NAME=OFF"
     "-DHDF4_ENABLE_JPEG_LIB_SUPPORT=ON"
     "-DHDF4_ENABLE_NETCDF=OFF"
-    "-DHDF4_ENABLE_SZIP_ENCODING=ON"
-    "-DHDF4_ENABLE_SZIP_SUPPORT=ON"
     "-DHDF4_ENABLE_Z_LIB_SUPPORT=ON"
     "-DHDF4_BUILD_FORTRAN=OFF"
     "-DJPEG_DIR=${libjpeg}"
+  ] ++ stdenv.lib.optionals (szip != null) [
+    "-DHDF4_ENABLE_SZIP_ENCODING=ON"
```

Comment:
@7c6f434c 

`Users who cannot legally use Szip encoding should download and build the HDF4 library with the decode-only Szip library.`

I am suggesting that we enable this code path by default and you are suggesting to leave out szip completely.

```    
{ stdenv, fetchurl }:

stdenv.mkDerivation {
  name = "szip-2.1";
  src = fetchurl {
    url = ftp://ftp.hdfgroup.org/lib-external/szip/2.1/src/szip-2.1.tar.gz;
    sha256 = "1vym7r4by02m0yqj10023xyps5b21ryymnxb4nb2gs32arfxj5m8";
  };

  meta = {
    description = "Compression library that can be used with the hdf5 library";
    homepage = http://www.hdfgroup.org/doc_resource/SZIP/;
    license = stdenv.lib.licenses.unfree;
  };
}
```
Let me first note that the description is incomplete, because it can be used with hdf4 and 5.

I am saying (all along) that this is too restrictive. Once this is fixed (by exposing an option to compile it only with decoding), *then* this package can be fixed as I detailed before and *sure* it's still non-free and indeed, Hydra would still not build it. 

Having said that, I strongly believe that Hydra would be allowed to distribute it when it is configured not to do compression, because it's clearly the intention of the licensing company to have a model like the company who developed MP3. They are just bad at writing down what they want. So, if someone would e-mail them, they would likely welcome people to distribute binary versions of it. If I was running a HPC cluster of this stuff, *I* would send the e-mail. 

Relevant (https://wiki.debian.org/DebianScience/Meteorology): 
```
SZIP is a compression library for HDF and other scientific data. HDF which we ship, can be built against it, but SZIP is patented (US only?) and is free for non-commercial use (ie. not DFSG-free). It is unlikely to change and we could not ship it (can it be shipped in non-free ?). However it would be desirable for HDF on Debian to be able to use it if present: hence a patch could be written to dynamically open it if present. The patent is owned by NASA who have agreed not to enforce it; it expires in a year or so. A GNU GPL'd version of SZIP is being written so that SZIP can be included in WMO standards.
```

Under the current legal requirement it's safest not to build szip in any form on Hydra, but that doesn't mean there shouldn't be options exposed for users to control what is being built without those users having to go through 4 layers to figure out where to patch something. 

In its current form using szip as packaged in NixPkgs is only possible for commercial users with a license.